excellent/functions/builtin.go

```diff
@@ -675,6 +676,48 @@ func Clean(env utils.Environment, text types.XText) types.XValue {
 	return types.NewXText(nonPrintableRegex.ReplaceAllString(text.Native(), ""))
 }
 
+// TextSlice returns the portion of `text` between `start` and `end` (inclusive).
```

Comment:
start is inclusive, end is exclusive, right?

Reply:
yep yep, updated.